code-challenges.test.js

```diff
@@ -17,36 +17,86 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-var secretCodeWord1 = "lackadaisical"
+// var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
-var secretCodeWord2 = "gobbledygook"
+// var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-
-
+describe("codedMessage", () => {
+    it("returns a coded message for an input of a string", () => {
+        const secretCodeWord1 = "lackadaisical";
+        const secretCodeWord2 = "gobbledygook";
+        expect( codedMessage(secretCodeWord1) ).toEqual("l4ck4d41s1c4l");
+        expect( codedMessage(secretCodeWord2) ).toEqual("g0bbl3dyg00k");
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
-
+// Create a function that takes in 1 argument that is a string
+codedMessage = (string) => {
+    // Split the string into an array
+    stringArr = string.split('');
+
+    // Use a higher-order function (.map) to convert a, e, i, o
+    codedArr = stringArr.map(value => {
+        if (value.toLowerCase() === 'a') {
+            return '4';
+        } else if (value.toLowerCase() === 'e') {
+            return '3';
+        } else if (value.toLowerCase() === 'i') {
+            return '1';
+        } else if (value.toLowerCase() === 'o') {
+            return '0';
+        } else {
+            return value;
+        }
+    });
+
+    // Join the array back into a string
+    // Return a string with the coded message
+    return codedArr.join('');
+}
 
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
 // a) Create a test with an expect statement using the variable provided.
 
-var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
+// var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
-
+describe("wordsWithLetterA", () => {
+    it("returns an array of wordds that contain the letter 'a'", () => {
+        const arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"];
+        const arrayOfWordsExpected = ["Apple", "Banana", "Peach"];
+        expect( wordsWithLetterA(arrayOfWords) ).toEqual(arrayOfWordsExpected);
+    })
+})
 
 
 
 // b) Create the function that makes the test pass.
 
-
-
+// Create a function that takes in an array of words
+const wordsWithLetterA = (array) => {
+    // Use a higher-order function (.filter) to get the words we want
+    return array.filter(value => {
+        // Check if the word contains the letter 'a'
+        if (value.toLowerCase().includes('a')) {
+            // Return true
+            return true;
+        } else {
+            // Otherwise, it does not contain the letter 'a'
+            // Return false
+            return false;
+        }
```

Comment:
The logic is right. Here's a refactoring suggestion:
```suggestion
    return array.filter(value => value.toLowerCase().includes('a'))
```

Because `.includes()` returns a boolean we don't need to explicitly code for returning the boolean